service/resource/configmap/service.go

```diff
@@ -72,9 +72,9 @@ type Service struct {
 }
 
 func (s *Service) GetCurrentState(ctx context.Context, obj interface{}) (interface{}, error) {
-	customObject, ok := obj.(*ingresstpr.CustomObject)
-	if !ok {
-		return nil, microerror.Maskf(wrongTypeError, "expected '%T', got '%T'", &ingresstpr.CustomObject{}, obj)
+	customObject, err := toCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err), nil
```

Comment:
This aligns the custom type handling using a helper method like in other operators.